core/source/api/opencmiss/zinc/fieldimage.h

```diff
@@ -531,7 +531,21 @@ ZINC_API enum cmzn_streaminformation_image_pixel_format
  * @return  Status CMZN_OK on success, any other value on failure.
  */
 ZINC_API int cmzn_field_image_set_pixel_format(cmzn_field_image_id image_field,
-	enum cmzn_streaminformation_image_pixel_format pixel_format);
+	enum cmzn_field_image_pixel_format pixel_format);
+
+/**
+ * Get the image buffer  currently in the image field texture. This is not a safe
+ * copy of the buffer and editing this buffer may result in error.
+ *
+ * @param image_field 	Handle to the image field.
+ * @param memory_buffer_reference  Will be set to point to the allocated
+ * 	memory block.
+ * @param memory_buffer_size  Will be set to the length of
+ * 	the returned memory block.
+ * @return  Status CMZN_OK on success, any other value on failure.
+  */
+ZINC_API int cmzn_field_image_get_buffer(cmzn_field_image_id image_field,
+	void **memory_buffer_references, unsigned int *memory_buffer_sizes);
```

Comment:
This should be const void ** if client is not expected to modify buffer.
The argument names don't match the doxygen comments and aren't great. Maybe change to:
const void **buffer_out, unsigned int *buffer_length_out?
Same problem in streamresource too, C and C++, SWIG typemaps. cmzn_streamresource_memory_get_buffer_copy should not be const.
Make sure docs say the length is in bytes.

Reply:
While I agree with you, changing the const will break all exisitng codes that uses this APIs.